src/js/common/helpers/login-helpers.js

```diff
@@ -93,6 +96,13 @@ export function getUserData(dispatch) {
     } else {
       dispatch(profileLoadingFinished());
     }
+  }).catch(error => {
```

Comment:
If we're going to add this `catch` block, you can probably remove the `json.data` check in the `then` above it, since that's the existing error handling (such as it is).

Reply:
Ah gotcha, good catch. It makes sense to me to separate the error handling this way. I haven't checked, but wouldn't the previous way have errored out since `undefined` is being passed to `then` on line 66 if `response.ok` is false? Doesn't really matter if we're going this route anyway but just something I noticed.

Reply:
With the `response.ok` check that you added, yeah it would throw an `undefined` error. I meant that the way it was before, without the `response.ok` check, the `json.data` check is essentially the same error check (our errors have json payloads, but they don't have `data` in them).

I agree that the catch block is better.

Reply:
Ah that makes sense, good to know. Thanks for the explanation!